### PR TITLE
[GSM Observability] Add mesh_id support in injected labels

### PR DESCRIPTION
**Pull request by yashykt**
Changes - 
1) Change local mesh labels to not be reported on 'started' metrics at all (even those that we know about) to be consistent. (Since xDS labels atleast on the server side would not be available on started metric.)
2) Add mesh_id as a local label that is populated by reading the xDS bootstrap. As part of this, also added a minimal xds bootstrap parsing logic.

### CODE REVIEW

File: src/cpp/ext/gsm/metadata_exchange.cc [hunk 1]
```diff
@@ -133,41 +204,25 @@ absl::string_view GetStringValueFromUpbStruct(google_protobuf_Struct* struct_pb,
...
+  explicit MeshLabelsIterable(
+      const std::vector<std::pair<absl::string_view, std::string>>&
+          local_labels,
+      grpc_core::Slice remote_metadata)
+      : local_labels_(local_labels), metadata_(std::move(remote_metadata)) {}
 
   absl::optional<std::pair<absl::string_view, absl::string_view>> Next()
       override {
     auto& struct_pb = GetDecodedMetadata();
+    auto local_labels_size = local_labels_.size();
```
Comment (markdroth) [inline src/cpp/ext/gsm/metadata_exchange.cc:218]: `auto` doesn't add any value here.  Please say `size_t` instead.

File: src/cpp/ext/gsm/metadata_exchange.h [hunk 2]
```diff
@@ -58,6 +54,11 @@ class ServiceMeshLabelsInjector : public LabelsInjector {
   grpc_core::Slice serialized_labels_to_send_;
 };
 
+// Returns the mesh ID by reading and parsing the bootstrap file. Returns
+// "unknown" if for some reason, mesh ID could not be figured out.
+// Exposed in header for test purposes.
+std::string GetMeshId();
```
Comment (markdroth) [inline src/cpp/ext/gsm/metadata_exchange.h:60]: Please add a comment saying that this is exposed for testing purposes only.

File: src/cpp/ext/gsm/metadata_exchange.cc [hunk 3]
```diff
@@ -56,6 +69,8 @@ constexpr absl::string_view kMetadataExchangeLocationKey = "location";
 constexpr absl::string_view kMetadataExchangeProjectIdKey = "project_id";
 constexpr absl::string_view kMetadataExchangeCanonicalServiceKey =
     "canonical_service";
+// The keys that will be used for the local attributes when recording metrics.
+constexpr absl::string_view kMeshIdAttribute = "gsm.mesh_id";
```
Comment (markdroth) [inline src/cpp/ext/gsm/metadata_exchange.cc:73]: In a separate PR, we need to change all occurrances of the string "gsm" to "csm".  This includes attribute names, file and directory names, BUILD target names, etc.

**Review by markdroth — APPROVED** [review]

File: src/cpp/ext/gsm/metadata_exchange.cc (hunk 3 above)
Comment (yashykt) [reply src/cpp/ext/gsm/metadata_exchange.cc:73]: Yes!

File: src/cpp/ext/gsm/metadata_exchange.h (hunk 2 above)
Comment (yashykt) [reply src/cpp/ext/gsm/metadata_exchange.h:60]: Already had it. Made it in caps now :)

File: src/cpp/ext/gsm/metadata_exchange.cc [hunk 4]
```diff
@@ -193,12 +249,12 @@ class PeerLabelsIterable : public LabelsIterable {
...
     if (struct_pb.struct_pb == nullptr) {
-      return 0;
+      return local_labels_.size();
     }
     if (type_ != GcpResourceType::kGke) {
-      return 1;
+      return local_labels_.size() + 1;
     }
-    return kGkeAttributeList.size();
+    return local_labels_.size() + kGkeAttributeList.size();
```
[the remark this answers is not in the document]
Comment (yashykt) [reply src/cpp/ext/gsm/metadata_exchange.cc]: Done

File: src/cpp/ext/gsm/metadata_exchange.cc (hunk 1 above)
Comment (yashykt) [reply src/cpp/ext/gsm/metadata_exchange.cc]: done

File: src/cpp/ext/gsm/metadata_exchange.h (hunk 2 above)
Comment (markdroth) [reply src/cpp/ext/gsm/metadata_exchange.h:60]: Whoops, sorry -- I swear I can read. :)

**Pull request merged**